clj-refactor.el

```diff
@@ -429,17 +429,31 @@ e.g. `re-search-forward'"
      (t (error "Only know how to search :forward or :backward, you asked for '%s'"
                direction)))))
 
+(defun cljr--point-for-anon-function ()
+  "Returns the location of point if the point is currently placed
+at the opening parentheses of an anonymous function."
+  (cond
+   ((looking-at "(fn \\(\\_<[^ ]+\\_>[[:space:]\n]+\\)?\\[")
+    (point))
+   ((save-excursion (backward-char) (looking-at "#("))
+    (1- (point)))))
+
 (defun cljr--goto-fn-definition ()
-  (let* ((search-bound (cljr--point-after 'cljr--goto-toplevel))
-         (literal (cljr--point-at-text-matching "#(" :backward search-bound
-                                                :noerror))
-         (fn (cljr--point-at-text-matching "(fn \\[" :backward search-bound
-                                           :noerror)))
-    (if (or literal fn)
-        (progn (goto-char (max (or literal (point-min)) (or fn (point-min))))
-               (when (looking-back "#")
-                 (backward-char)))
-      (error "Can't find definition of anonymous function!"))))
+  (if (zerop (first (paredit-current-parse-state)))
+      (error "Not inside a s-expression.")
+    (let* ((pt-orig (point))
+           (search-bound (cljr--point-after 'cljr--goto-toplevel))
+           found-fn-p)
+      (while (not found-fn-p)
```

Comment:
Won't this loop forever if there's no fn definition between here and the toplevel of the file?  I think maybe this needs to be guarded with a check on `cljr--toplevel-p?`